projects/spectator/src/lib/mock.ts

```diff
@@ -3,7 +3,7 @@ import { FactoryProvider, Type, AbstractType } from '@angular/core';
 
 type Writable<T> = { -readonly [P in keyof T]: T[P] };
 
-export declare type UnknownFunction = (...args: Array<unknown>) => unknown & Function;
```

Comment:
`& Function` causes the type checks in Typescript to not match correctly. 

The new definition is "function like" which is good enough for the type check requirements.